src/apify/_memory_storage/resource_clients/dataset.py

```diff
@@ -10,7 +10,7 @@
 from apify_shared.types import JSONSerializable
 from apify_shared.utils import ignore_docs
 
-from ..._crypto import _crypto_random_object_id
+from ..._crypto import crypto_random_object_id
```

Comment:
I believe that going that many levels up with relative imports is counter-productive. I suggest replacing those with absolute imports.